docs/tests.md

```diff
@@ -0,0 +1,81 @@
+
+## cdperf Tests
+
+This lists and briefly explains the tests that are done in the K6 based test framework cdperf.
+
+### Most used file sizes upload
+
+Script: `/root/cdperf/tests/k6/test-issue-github-enterprise-4115-most-used-sizes-upload.js`
+
+This test creates a pattern of the so called *most used file sizes*. It contains a collection of test files in sizes how we find them in typical larger ownCloud installations. Specifically, that contains 84% files with a size smaller than 1MB, 12.5% files between 1MB and 10MB, 3% files between 1MB and 100MB and 0.8% files larger than 100MB.
+
+The files are created, uploaded, downloaded and afterwards deleted in this test. The current implementation tests with in total 15GB data per User.
+
+This simulates a "normal" upload/download scenario with a common file set.
+
+### Deep Propfind
+
+Script: `/root/cdperf/tests/k6/test-issue-github-ocis-1018-propfind-deep.js`
+
+This test creates a nested directory structure that is fife levels deep and distributes 1000 files within that tree.
+
+The files are created and uploaded. After that, one PROPFIND on the top directory is done.
+
+This simulates the ETag check of a large nested directory.
+
+### Flat Propfind
+
+Script: `/root/cdperf/tests/k6/test-issue-github-ocis-1018-propfind-flat.js`
+
+This test crates many (1000 a 1kB) files in the root of the cloud.
+
+The files are uploaded and afterwards, a PROPFIND to the root directory is done.
+
+This test gives an indication on the PROPFIND speed of a large root and also measures the pure upload speed of small files.
+
+### Upload Big Files
+
+Script: `/root/cdperf/tests/k6/test-issue-github-ocis-1018-upload-download-delete-many-large.js`
+
+This test creates in sum 5 GB of files larger than 5 MB and up to 1GB.
```

Comment:
In total this test creates 5gb of files which distributes from 5mb up to 1gb per file